File: backend/src/routes/testRoute.js
const express = require('express');
const mongoose = require('mongoose');

const router = express.Router();

router.get('/', async (_req, res) => {
  const statusMap = {
    0: 'disconnected',
    1: 'connected',
    2: 'connecting',
    3: 'disconnecting',
  };

  const dbStatus = statusMap[mongoose.connection.readyState] || 'unknown';

  res.json({
    message: 'Test endpoint is working',
    database: dbStatus,
    timestamp: new Date().toISOString(),
  });
});

module.exports = router;

File: backend/src/routes/testRoute.py
from __future__ import annotations

from datetime import datetime

from fastapi import APIRouter

from ..db import _client as db_client  # type: ignore[attr-defined]

router = APIRouter()


@router.get("/")
async def test_endpoint():
    status = "connected" if db_client is not None else "disconnected"
    return {
        "message": "Test endpoint is working",
        "database": status,
        "timestamp": datetime.utcnow().isoformat(),
    }
